internal/cri/server/images/image_pull.go

```diff
@@ -117,15 +165,42 @@ func (c *GRPCCRIImageService) PullImage(ctx context.Context, r *runtime.PullImag
 	if err != nil {
 		return nil, err
 	}
+	// Trace: returning normalized image ref to caller
 	return &runtime.PullImageResponse{ImageRef: ref}, nil
 }
 
 func (c *CRIImageService) PullImage(ctx context.Context, name string, credentials func(string) (string, string, error), sandboxConfig *runtime.PodSandboxConfig, runtimeHandler string) (_ string, err error) {
-	span := tracing.SpanFromContext(ctx)
+	ctx, span := tracing.StartSpan(ctx, tracing.Name("cri.image", "pull"))
```

Comment:
probably don't need a new span here..  I believe it would be more important to make sure the paths that get here have already set the span in the passed in ctx.  

Example.. run pod sandbox creating a core internal sandboxer/sandbox will ensure the sandbox pause container image is local..